Provide actionable guidance on how to use this document (#117)
oauth-wg/oauth-cross-device-security: draft-ietf-oauth-cross-device-security.md

These attacks borrow techniques from traditional phishing attacks, but instead of collecting passwords, attackers collect session transfer codes and other artefacts that allow them to setup a session and then use it to access a user's data.

## Defending Against Cross-Device Attacks
This document provides guidance to implementers to defend against Cross-Device Consent Phishing and Cross-Device Session Phishing attacks. This guidance includes:

> 1. Practical mitigations for suceptible protocols ((#practical-mitigations)).

Reviewer: ```suggestion
1. Practical mitigations for susceptible protocols ((#practical-mitigations)).
```

2. Protocol selection guidance to avoid using susceptible protocols ((#protocol-selection)).

> 3. Results from formal analysis of suceptible protocols ((#foundational-pillars)).

Reviewer: ```suggestion
3. Results from formal analysis of susceptible protocols ((#foundational-pillars)).
```

"authorization request", and
"client" defined by The OAuth 2.0 Authorization Framework [@!RFC6749].

# Best Practices

> This section describes the set of security mechanisms and measures to secure cross-device protools against Cross-Device Cosnsent Phishing and Cross-Device Session Phishing attacks that the OAuth working group considers best practices at the time of writing.

Reviewer: ```suggestion
This section describes the set of security mechanisms and measures to secure cross-device protools against Cross-Device Consent Phishing and Cross-Device Session Phishing attacks that the OAuth working group considers best practices at the time of writing.
```

1. Implementers MUST perform a risk assessment before implementing cross-device flows, weighing the risks from Cross-Device Cosnsent Phishing and Cross-Device Session Phishing attacks against benefits for users.
2. Implementers SHOULD avoid cross-device flows if risks cannot be sufficiently mitigated.

> 3. Implementers must follow the guidance provided in (#protocol-selection) for protocol selection.

Reviewer: ```suggestion
3. Implementers SHOULD follow the guidance provided in (#protocol-selection) for protocol selection.
```

> 4. Implementers MUST implement practical mitigations as listed in (#practical-mitigations) that is appropriate for the use case, architecture, and selected protocols.

Reviewer: ```suggestion
4. Implementers MUST implement practical mitigations as listed in (#practical-mitigations) that are appropriate for the use case, architecture, and selected protocols.
```

5. Implementers SHOULD implement proximity checks as defined in (#establish-proximity) if possible.

> These best practices apply to the Device Authorization Grant ([@RFC8628]) as well as other cross-device protocols such as the Client Initiated Backchannel Authentication [@CIBA], Self-Issued OpenID Provider v2 [@OpenID.SIOPV2], the Pre-Authorized Code Flow in ([@OpenID.VCI]) and other cross-device protocols that rely on the user to authenticate the channel between devices.

Reviewer: In line with the other PR, do you like to mention VP here?

Reviewer: ```suggestion
These best practices apply to the Device Authorization Grant ([@RFC8628]) as well as other cross-device protocols such as the Client Initiated Backchannel Authentication [@CIBA], Self-Issued OpenID Provider v2 [@OpenID.SIOPV2], OpenID for Verifiable Presentations [@OpenID.VCP], the Pre-Authorized Code Flow in ([@OpenID.VCI]) and other cross-device protocols that rely on the user to authenticate the channel between devices.
```

> These best practices applies to the Device Authorization Grant ([@RFC8628]) as well as other cross-device protocols such as the Client Initiated Backchannel Authentication [@CIBA], Self-Issued OpenID Provider v2 [@OpenID.SIOPV2], the Pre-Authorized Code Flow in ([@OpenID.VCI]) and other cross-device protocols that relies on the user to authenticate the channel between devices.

Reviewer: ```suggestion
These best practices apply to the Device Authorization Grant ([@RFC8628]) as well as other cross-device protocols such as the Client Initiated Backchannel Authentication [@CIBA], Self-Issued OpenID Provider v2 [@OpenID.SIOPV2], the Pre-Authorized Code Flow in ([@OpenID.VCI]) and other cross-device protocols that rely on the user to authenticate the channel between devices.
```

   -latest

>    * Added section to provide actionable guidance to implementors on how to use this document.

Reviewer: ```suggestion
   * Added section to provide actionable guidance to implementers on how to use this document.
```
